python/nav/web/templates/maintenance/calendar.html

```diff
@@ -11,8 +11,8 @@
       </a>
 
       <form action="{% url 'maintenance-calendar' %}" method="get" class="inline-form">
-        <input type="text" name="year" value="{{ this_month.year }}" size="4" maxlength="4"/>
-        <input type="text" name="month" value="{{ this_month.month }}" size="2" maxlength="2"/>
+          <input type="number" name="month" value="{{ this_month.month }}" min="1" max="12" required id="id_month"/>
+          <input type="number" name="year" value="{{ this_month.year }}" min="2000" max="2100" required id="id_year"/>
```

Comment:
And here's a few suggestions:

1. The month value should now be taken from `calendarform.month.value`, and the year value from `calendarform.year.value`
2. Defining the value limits both in the form code and in the HTML template is redundant. If we want to change them, we now need to do it in *2* places. Better to use something like `min="{{ calendarform.month.min_value }}"` and `max="{{ calendarform.month.max_value }}"`, where the limits only need to be defined in the form class.
3. This form still does not display validation errors to the end user (if they somehow managed to input invalid data, something only malicious/curious users would be able to do here). Have a look at these examples to see how you can render form fields and errors manually in templates:

https://docs.djangoproject.com/en/3.2/topics/forms/#rendering-fields-manually